frontend/public/components/modals/impersonate-user-modal.tsx

```diff
@@ -0,0 +1,357 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  Alert,
+  AlertVariant,
+  Popover,
+  Select,
+  SelectList,
+  SelectOption,
+  MenuToggle,
+  MenuToggleElement,
+  Label,
+  Badge,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons/dist/esm/icons/help-icon';
+import { TimesIcon } from '@patternfly/react-icons/dist/esm/icons/times-icon';
+import { Modal, ModalVariant } from '@patternfly/react-core/deprecated';
+
+export interface ImpersonateUserModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onImpersonate: (username: string, groups: string[]) => void;
+  prefilledUsername?: string;
+  isUsernameReadonly?: boolean;
+}
+
+export const ImpersonateUserModal: React.FC<ImpersonateUserModalProps> = ({
+  isOpen,
+  onClose,
+  onImpersonate,
+  prefilledUsername = '',
+  isUsernameReadonly = false,
+}) => {
+  const { t } = useTranslation();
+  const [username, setUsername] = React.useState(prefilledUsername);
+  const [selectedGroups, setSelectedGroups] = React.useState<string[]>([]);
+  const [usernameError, setUsernameError] = React.useState('');
+  const [isGroupSelectOpen, setIsGroupSelectOpen] = React.useState(false);
+  const [showAllGroups, setShowAllGroups] = React.useState(false);
+  const [groupSearchFilter, setGroupSearchFilter] = React.useState('');
+
+  // Mock group options - in real implementation, these would come from API
```

Comment:
To be honest if we have to mock API in order for this PR to work I would put WIP in the title